File: app/src/main/java/at/shockbytes/corey/util/schedule/ScheduleManager.java
package at.shockbytes.corey.util.schedule;

import java.util.List;

import at.shockbytes.corey.storage.live.LiveScheduleUpdateListener;
import io.reactivex.Observable;

/**
 * @author Martin Macheiner
 *         Date: 21.02.2017.
 */

public interface ScheduleManager {

    void poke();

    Observable<List<ScheduleItem>> getSchedule();

    Observable<List<String>> getItemsForScheduling();

    ScheduleItem insertScheduleItem(ScheduleItem item);

    void updateScheduleItem(ScheduleItem item);

    void deleteScheduleItem(ScheduleItem item);

    boolean isWorkoutNotificationDeliveryEnabled();

    boolean isWeighNotificationDeliveryEnabled();

    int getDayOfWeighNotificationDelivery();

    void postWeighNotification();

    void tryPostWorkoutNotification();

    void registerLiveForScheduleUpdates(LiveScheduleUpdateListener listener);

    void unregisterLiveForScheduleUpdates(LiveScheduleUpdateListener listener);

}

File: app/src/main/java/at/shockbytes/corey/schedule/ScheduleManager.kt
package at.shockbytes.corey.schedule

import at.shockbytes.corey.common.core.util.Pokeable
import io.reactivex.Observable

/**
 * @author Martin Macheiner
 * Date: 21.02.2017.
 */

interface ScheduleManager : Pokeable {

    val schedule: Observable<List<ScheduleItem>>

    val itemsForScheduling: Observable<List<String>>

    val isWorkoutNotificationDeliveryEnabled: Boolean

    val isWeighNotificationDeliveryEnabled: Boolean

    val dayOfWeighNotificationDelivery: Int

    fun insertScheduleItem(item: ScheduleItem): ScheduleItem

    fun updateScheduleItem(item: ScheduleItem)

    fun deleteScheduleItem(item: ScheduleItem)

    fun postWeighNotification()

    fun postWorkoutNotification()

    fun registerLiveScheduleUpdates(listener: LiveScheduleUpdateListener)

    fun unregisterLiveScheduleUpdates()

}
